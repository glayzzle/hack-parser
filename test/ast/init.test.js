const parser = require('../../src/index');

describe('simple syntax tests', function() {
  it('parser should pass', function() {
    expect(parser.parseCode(`<?hh

      <<__Memoize>>
      function foo(inout vec<string> $bar): {
        /*using (new Baz()) {
        }*/
      }
    `)).toMatchSnapshot();
  });

});
