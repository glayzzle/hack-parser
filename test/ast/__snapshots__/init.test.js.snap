// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple syntax tests parser should pass 1`] = `
Program {
  "children": Array [
    _Function {
      "arguments": Array [
        Parameter {
          "byref": false,
          "inout": true,
          "kind": "parameter",
          "modifiers": null,
          "name": "bar",
          "nullable": false,
          "type": Identifier {
            "arguments": Array [
              Identifier {
                "kind": "identifier",
                "name": "string",
                "resolution": "uqn",
              },
            ],
            "kind": "identifier",
            "name": "vec",
            "resolution": "rn",
          },
          "value": null,
          "variadic": false,
        },
      ],
      "attributes": Array [
        Attribute_Spec {
          "args": null,
          "kind": "attribute_spec",
          "name": "__Memoize",
        },
      ],
      "body": Block {
        "children": Array [],
        "kind": "block",
      },
      "byref": false,
      "kind": "function",
      "name": "foo",
      "nullable": false,
      "type": null,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
